segmentstore/server/src/main/java/io/pravega/segmentstore/server/attributes/SegmentAttributeBTreeIndex.java

```diff
@@ -671,29 +704,47 @@ private void storeInCache(List<Map.Entry<Long, ByteArraySegment>> toAdd, Collect
 
             // Add new ones.
             for (val e : toAdd) {
-                long offset = e.getKey();
-                ByteArraySegment data = e.getValue();
-                CacheEntry entry = this.cacheEntries.getOrDefault(offset, null);
-                if (entry == null || entry.getSize() != data.getLength()) {
-                    entry = new CacheEntry(offset, data.getLength(), this.currentCacheGeneration);
-                    this.cacheEntries.put(offset, entry);
-                }
-
-                storeInCache(entry, data);
+                storeInCache(e.getKey(), e.getValue());
             }
         }
     }
 
     @GuardedBy("cacheEntries")
-    private void storeInCache(CacheEntry entry, ByteArraySegment data) {
-        int newAddress;
+    private void storeInCache(long entryOffset, ByteArraySegment data) {
+        CacheEntry entry = this.cacheEntries.getOrDefault(entryOffset, null);
+        if (entry != null && entry.getSize() == data.getLength()) {
+            // Already cached.
+            return;
```

Comment:
Codecov complains this line is not covered.
Seems like an important case to cover.

Reply:
I can't cover that case anymore since we only issue one read request per page at any given time (#5406) . There is no chance that two read requests for the same page execute in parallel. 

This is old code that was moved around.